controllers/btpoperator_controller.go

```diff
@@ -424,75 +220,178 @@ func (r *BtpOperatorReconciler) HandleProcessingState(ctx context.Context, cr *v
+		logger.Error(err, "while getting the required Secret")
+		return nil, NewErrorWithReason(MissingSecret, "Secret resource not found")
+	}
+
+	logger.Info("verifying the required Secret")
+	if err = r.verifySecret(secret); err != nil {
+		logger.Error(err, "while verifying the required Secret")
+		return nil, NewErrorWithReason(InvalidSecret, "Secret validation failed")
+	}
+	return secret, nil
+}
+
+func (r *BtpOperatorReconciler) getRequiredSecret(ctx context.Context) (*corev1.Secret, error) {
+	secret := &corev1.Secret{}
+	objKey := client.ObjectKey{Namespace: ChartNamespace, Name: SecretName}
+	if err := r.Get(ctx, objKey, secret); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, fmt.Errorf("%s Secret in %s namespace not found", SecretName, ChartNamespace)
 		}
-		if err = r.reconcileResources(ctx, us); err != nil {
-			logger.Error(err, "while provisioning resources")
-			return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, ProvisioningFailed, "Failed to provision module resources")
+		return nil, fmt.Errorf("unable to get Secret: %w", err)
+	}
+
+	return secret, nil
+}
+
+func (r *BtpOperatorReconciler) verifySecret(secret *corev1.Secret) error {
+	missingKeys := make([]string, 0)
+	missingValues := make([]string, 0)
+	errs := make([]string, 0)
+	requiredKeys := []string{"clientid", "clientsecret", "sm_url", "tokenurl", "cluster_id"}
+	for _, key := range requiredKeys {
+		value, exists := secret.Data[key]
+		if !exists {
+			missingKeys = append(missingKeys, key)
+			continue
 		}
-		if err = r.waitForResourcesReadiness(ctx, us); err != nil {
-			logger.Error(err, "while waiting for resources readiness")
-			return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, ProvisioningFailed, "Timed out while waiting for resources readiness")
+		if len(value) == 0 {
+			missingValues = append(missingValues, key)
 		}
 	}
+	if len(missingKeys) > 0 {
+		missingKeysMsg := fmt.Sprintf("key(s) %s not found", strings.Join(missingKeys, ", "))
+		errs = append(errs, missingKeysMsg)
+	}
+	if len(missingValues) > 0 {
+		missingValuesMsg := fmt.Sprintf("missing value(s) for %s key(s)", strings.Join(missingValues, ", "))
+		errs = append(errs, missingValuesMsg)
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("%s", strings.Join(errs, ", "))
+	}
 
-	logger.Info("provisioning succeeded")
-	return r.UpdateBtpOperatorStatus(ctx, cr, types.StateReady, ReconcileSucceeded, "Module provisioning succeeded")
+	return nil
 }
 
-func (r *BtpOperatorReconciler) createUnstructuredObjectsFromManifestsDir(ctx context.Context, manifestsDir string) ([]*unstructured.Unstructured, error) {
+func (r *BtpOperatorReconciler) deleteOutdatedResources(ctx context.Context) error {
 	logger := log.FromContext(ctx)
 
+	logger.Info("getting outdated module resources to delete")
+	usToDelete, err := r.createUnstructuredObjectsFromManifestsDir(r.getResourcesToDeletePath())
+	if err != nil {
+		logger.Error(err, "while getting deletable objects from manifests")
```

Comment:
Those objects are to be deleted and all objects in manifest are deletable = can be deleted.

Reply:
changed to `while getting objects to delete from manifests`